### PR TITLE
chore(deps): remove react-rectangle-selection (#18017)

### LINKED ISSUE #18017: chore(deps): remove react-rectangle-selection
## Description

`react-rectangle-selection` powers the drag-a-rectangle selection mode of the 2D graph (knowledge graphs, investigations, correlations). It is wrapped by `opencti-platform/opencti-front/src/components/graph/components/RectangleSelection.tsx` and consumed from `Graph.tsx`. That is its only use in the platform.

The package was last published in **January 2019** (`1.0.4`, ISC) and has since become the most awkward dependency of the front.

### It declares `react` and `webpack` as runtime dependencies

```json
"dependencies": {
  "react": "^16.7.0",
  "webpack": "^4.28.4"
}
```

Both belong in `peerDependencies` / `devDependencies`. `react-rectangle-selection` is the only package in the entire `opencti-front` dependency graph that declares `react` as a hard dependency, and the only one that depends on `webpack` at all:

```console
$ awk '/^"[^"]+@npm/{pkg=$0} /    webpack: "npm:/{print pkg" -> "$0}' yarn.lock
"react-rectangle-selection@npm:1.0.4": ->     webpack: "npm:^4.28.4"
```

The front is a Vite application. It nevertheless installs **9.7 MB of webpack 5 and its 20 direct dependencies**, none of which is ever loaded, and `package.json` carries two `resolutions` entries whose only purpose is to neutralise this single package:

```json
"resolutions": {
  "react": "19.2.8",
  "webpack": "5.109.2"
}
```

Without the `webpack` override, the tree resolves to webpack 4 and its unmaintained transitive chain. With it, we silently force a major version bump onto a package that nobody builds with. Neither is a defensible state; the dependency should not be there at all. The same applies to the `react` override, which exists to stop React 16 from being installed side by side with React 19.

### It ships no types

Hence `declare module 'react-rectangle-selection';` in `opencti-platform/opencti-front/index.d.ts`. The whole surface is `any`, so nothing about the props we pass is checked.

### Most of what it provides is already dead code in OpenCTI

The library renders its selection box using `pageX`/`pageY` directly as CSS `left`/`top`. That is only correct when the container sits at the document origin, which never happens behind the left navigation and the top bar. `RectangleSelection.tsx` therefore hides the library's own box with `style={{ display: 'none' }}` and re-implements the overlay, the canvas offset and the scroll compensation by hand.

What we still consume from the package is roughly 40 lines of `mousedown` / `mousemove` / `mouseup` bookkeeping inside a React 16 class component.

## Why swapping in another third-party library is not the right approach

The obvious reflex is to look for a maintained equivalent. The candidates are `@air/react-drag-to-select`, `react-selecto`, `dragselect` and `@viselect/react`. None of them is a good answer here.

**Their core feature is unusable in our case.** All of these libraries are *DOM element* selection engines: their value is the hit-testing pass that determines which child elements intersect the drawn box. Our graph is a single `<canvas>`. There are no DOM children to hit-test, and selection is resolved in graph space through `ForceGraphMethods.screen2GraphCoords`, exactly as `LassoSelection` and `RelationSelection` already do. We would import an entire selection engine to use its pointer-event tracking and throw the rest away — which is precisely the situation we are in today.

**Each candidate brings its own cost.** `react-selecto` pulls 10 transitive dependencies and has not been released since December 2023. `dragselect` is GPL-3.0 and would be rejected by `yarn verify-licenses`, whose allowlist stops at LGPL-3.0. `@viselect/react` is well maintained but sits around 12k weekly downloads — the same bus-factor profile we are trying to move away from. Swapping one micro-dependency for another buys a new upgrade treadmill around code we would still have to wrap.

**The in-house implementation already exists.** The two sibling selection modes in the same directory, `LassoSelection.tsx` (freehand polygon) and `RelationSelection.tsx`, are written in-house on top of the same graph API. The rectangle is the simplest of the three and the only one delegated to a package. Our wrapper already hides the library's rendering, computes the offsets, converts the coordinates and draws the overlay: what is left to write is the pointer bookkeeping.

## Proposed solution

Remove the dependency and fold the remaining logic into the existing wrapper:

- Rewrite `RectangleSelection.tsx` as a self-contained function component — pointer handlers on the container, `origin` / `target` state, and the overlay and coordinate logic it already owns.
- Keep the exported `RectangleSelectionProps` contract identical, so `Graph.tsx` and `useGraphInteractions.ts` are untouched.
- Position the selection box from container-relative coordinates, fixing the `pageX`/`pageY` bug at its source instead of compensating for it downstream.
- Drop `react-rectangle-selection` from `package.json`, drop the now-unnecessary `webpack` resolution, and re-evaluate the `react` one.
- Drop `declare module 'react-rectangle-selection'` from `index.d.ts`.
- Cover the component with unit tests, which the current `any`-typed wrapper has none of.

### Expected outcome

| | Before | After |
|---|---|---|
| Dependencies | `react-rectangle-selection` | — |
| `webpack` in `node_modules` | 9.7 MB, 20 direct deps | absent |
| `resolutions` entries | `react`, `webpack` | `react` under review |
| Untyped module declarations | 1 | 0 |
| Behaviour | unchanged | unchanged, positioning bug fixed |

### PR DESCRIPTION
**Pull request by xfournet**
### Proposed changes

* Remove `react-rectangle-selection` and fold the pointer bookkeeping it still provided into `RectangleSelection.tsx`. The wrapper already hid the library's own selection box and re-implemented the overlay and the coordinate conversion, so this deletes a dependency, not a feature.
* Drop the `webpack` and `react` entries from `resolutions`. Both existed only to neutralise this package: it was the sole dependant on `webpack` in the whole front, and the only one declaring `react` as a hard dependency. `webpack` leaves `node_modules` entirely (9.7 MB, 20 direct dependencies, −430 lines of `yarn.lock`); the lockfile is byte-identical with and without the `react` resolution once the package is gone.
* Drop `declare module 'react-rectangle-selection'` from `index.d.ts` — the component is now typed instead of being an opaque `any`.
* Derive the rectangle from `clientX`/`clientY` against the canvas bounding box. Both are viewport-relative, so the scroll compensation the wrapper needed to work around the library's `pageX`/`pageY` positioning disappears.
* Normalise the rectangle inside the component, so consumers always receive an ordered top-left / bottom-right pair.
* Start a drag only when the mousedown target is the graph canvas: the wrapper also holds the toolbar, the loading alert and the details bar, and dragging from those used to start a selection. The target is used directly, which removes the `document.querySelector()` lookup — and with it the `graphId` prop and the `graph-${uuid()}` identifier that `Graph.tsx` rebuilt on every render for that single consumer.
* Track the drag with document-level `mousemove` / `mouseup` listeners, the same pattern as `LassoSelection` and `RelationSelection`. The rectangle keeps following the cursor past the edges of the graph instead of being committed and truncated on `mouseleave`, and a release outside the graph applies the full rectangle.
* Two smaller fixes that came for free: a drag no longer starts on a non-primary mouse button, and text selection is suppressed while dragging.
* Add unit tests for the component, which had none.

The exported `RectangleSelectionProps` contract is otherwise unchanged, so `useGraphInteractions.ts` is untouched.

Why not another library: the maintained candidates (`@air/react-drag-to-select`, `react-selecto`, `dragselect`, `@viselect/react`) are all *DOM element* selection engines, and their hit-testing pass is unusable here — the graph is a single `<canvas>` and selection is resolved in graph space via `screen2GraphCoords`. We would import a whole engine for its pointer tracking and discard the rest, which is exactly the situation being removed. The rationale is detailed in the issue.

### Related issues

* closes #18017

### How to test this PR

1. Open any knowledge graph, investigation or correlation view in 2D.
2. Activate **Free rectangle select** in the graph toolbar.
3. Drag a rectangle over some nodes and release: the dashed overlay must follow the cursor and the enclosed nodes must end up selected.
4. Drag in each of the four directions (including bottom-right → top-left): the selection must be identical whatever the direction.
5. Hold <kbd>Shift</kbd> or <kbd>Alt</kbd> while dragging: the new nodes must be added to the current selection instead of replacing it.
6. Drag past the edges of the graph and release outside it: the rectangle must keep following the cursor and the full area must be applied.
7. Drag starting from the toolbar, the loading alert or the entity details bar: no rectangle must appear and those controls must stay usable.
8. Scroll the page (e.g. on a narrow window with the side panel open) and repeat: the rectangle must stay under the cursor.
9. Deactivate the mode: dragging must pan the graph again.

Regression coverage: `src/components/graph/components/RectangleSelection.test.tsx` (12 cases).

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant use cases (coverage and e2e)
- [ ] I added/updated the relevant documentation (either on GitHub or on Notion)
- [x] Where necessary, I refactored code to improve the overall quality

### Further comments

`yarn build`, `yarn check-ts`, `yarn lint` and `yarn test` are green. The single failing test in the suite, `src/utils/Time.test.ts > minutesBefore`, is timezone-dependent and fails identically on `master`.

### CODE REVIEW

File: opencti-platform/opencti-front/src/components/graph/components/RectangleSelection.tsx [hunk 1]
```diff
@@ -21,85 +20,111 @@ export interface RectangleSelectionProps {
...
+    return [clientX - left, clientY - top];
+  };
+
+  const startSelection = (event: MouseEvent) => {
+    if (disabled || event.button !== 0) return;
+    const canvas = document.querySelector(`#${graphId} canvas`);
+    if (!canvas) return;
+    canvasRect.current = canvas.getBoundingClientRect();
+    dragOrigin.current = canvasPoint(event);
+  };
```
[the remark this answers is not in the document]
Comment (xfournet) [reply opencti-platform/opencti-front/src/components/graph/components/RectangleSelection.tsx]: Good catch, fixed in dd320ad30d.

`startSelection` now bails out unless the mousedown target is the graph canvas, and uses that target directly, so the `document.querySelector()` is gone:

```ts
const canvas = event.target;
if (!(canvas instanceof HTMLCanvasElement)) return;
canvasRect.current = canvas.getBoundingClientRect();
```

That lookup was the only consumer of the `graphId` prop, which was itself the only consumer of the `graph-${uuid()}` identifier that `Graph.tsx` rebuilt on every render. Both are dropped, along with the DOM id they existed for.

I also took the related point about `onMouseLeave` from the review summary: tracking moved to document-level `mousemove` / `mouseup` listeners, the same pattern as `LassoSelection` and `RelationSelection`. The rectangle now keeps following the cursor past the edges of the graph instead of being committed and truncated on leave, and a release outside the graph applies the full rectangle. The listeners are released on unmount.

Covered by two new cases in `RectangleSelection.test.tsx` (12 in total): a mousedown on a non-canvas child starts nothing, and a drag running outside the graph reports the full rectangle.

**Review by fellowseb — APPROVED** [review]
Tested 👍
